Manak_Leah_ndvi_2022.py

```diff
@@ -0,0 +1,670 @@
+    date_location = directory_path[datetime_location[0]: datetime_location[1]]
+    format = "%Y%m%d"
+
+    # Create a date varaiable using new object and the datetime format
+    date = datetime.strptime(date_location, format)
+
+    # Create a location variable called "site"
+    site = directory_path[sitename_location[0]: sitename_location[1]]
+
+    # Append site and date variables to list
+    sitename_date.append(site)
+    sitename_date.append(date)
+    
+    # Return the populated sitename_date list
+    return sitename_date
+
+# A function to open clean landsat bands
+def open_clean_bands(band_path,
+                     crop_extent,
+                     valid_range=None):
+    """Open, crop, and identify the range of the bands.
+
+    Parameters
+    -----------
+    band_path : string
+        A path to the array that we will open
+    valid_range : tuple (optional)
+        A tuple of min and max range of values for the data. Default = None
+
+    Returns
+    -----------
+    arr : xarray DataArray
+        An xarray DataArray with values that should be masked
+        set to 1 for True (Boolean)
+    """
+    # tests to ensure the arrays are clipped to the same .shape
+    band = rxr.open_rasterio(
+        band_path, masked=True).rio.clip(
+        crop_extent.geometry,from_disk=True).squeeze()
+
+    # This last step is only for a valid tuple
+    if valid_range:
+        mask = ((band < valid_range[0]) | (band > valid_range[1]))
+        band = band.where(~xr.where(mask, True, False))
+
+    return band
+
+
+# In[6]:
+
+
+# Create dataframe of mean NDVI in this cell using the functions created above
+
+# Create path to the two sites "SJER" and "HARV"
+site_paths = glob(os.path.join("ndvi-automation",
+                               "sites",
+                               "*"))
+site_paths
+
+# Create cloud mask and get the cloud pixel values from earthpy
+high_cloud_confidence = (
+    em.pixel_flags["pixel_qa"]["L8"]["High Cloud Confidence"])
+cloud = em.pixel_flags["pixel_qa"]["L8"]["Cloud"]
+cloud_shadow = em.pixel_flags["pixel_qa"]["L8"]["Cloud Shadow"]
+
+all_masked_values = cloud_shadow + cloud + high_cloud_confidence
+harv_path = glob(os.path.join("ndvi-automation", "sites", "HARV"))
+
+# Open and clean all HARV: first create empty HARV list
+harv_info = []
+
+# Create a for loop for the HARV path
+for path in harv_path:
+    # Establish the scene directory path 
+    scene_path = glob(os.path.join(path, "landsat-crop",
+                      "LC080130302017031701T1-SC20181023151837"))
+    # Set the path to the cropped shapefile and open as the HARV boundary
+    bound = os.path.join(path, "vector", "HARV-crop.shp")
+    harv_boundary = gpd.read_file(bound)
+
+    # Create a nested for loop associated with each .tif file (band 4-5)
+    for tifs in scene_path:
+        # Get site and date info from the scene directory path
+        site_info = extract_date_sitename(tifs, [22, 26], [50, 58])
```

Comment:
@LManak for other people who aren't familiar with these values, it'd be great to comment why you chose these numbers